IndomitableIbises/Bangazon-API-Sprint-1
# changed delete methods

## Description of Proposed Changes

- delete methods now make sure that cascading deletion and protected data meet requirements

## Steps to Test
```sh
git fetch --all
git checkout SI_OnDelete
```

kill your DB and remake it
checkout all the featured data sets
ensure Dpt, customer, and emp are still not available for delete
check to see that payment_type deletion will cause set_null to occur on Order (I had to add the FK first)
Emp_Train will be deleted if attached event FK entry is deleted in Train table

## Impacted Areas in Application

Models.py

## Mentions @username

@rafcevallos @CashewRose @jcarter0149 

## Definition of Done

1. The project must be fully documented. This includes the following:
   1. Complete README that documents the steps to install the code, how to install any dependencies, or system configuration needed.
   1. Every class must be documented with purpose, author, and methods.
   1. Every method must be documented with purpose and argument list - which itself must contain a short purpose for each argument.
1. The project must be able to run fully, and without errors, on each teammate's system.
1. Fulfills every requirement.
Every line of code has been peer reviewed.
1. For projects that require unit testing, core functionality must be identified and have at least one test for each
